x/pairing/keeper/msg_server_freeze.go

```diff
@@ -0,0 +1,35 @@
+package keeper
+
+import (
+	"context"
+	"math"
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	"github.com/lavanet/lava/x/pairing/types"
+)
+
+func (k msgServer) Freeze(goCtx context.Context, msg *types.MsgFreeze) (*types.MsgFreezeResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	providerAddr, err := sdk.AccAddressFromBech32(msg.GetCreator())
+	if err != nil {
+		return nil, utils.LavaFormatError("Freeze_get_provider_address", err, &map[string]string{"providerAddress": msg.GetCreator()})
+	}
+
+	for _, chainId := range msg.GetChainIds() {
+		stakeEntry, found, index := k.epochStorageKeeper.GetStakeEntryByAddressCurrent(ctx, epochstoragetypes.ProviderKey, chainId, providerAddr)
+		if !found {
+			continue
```

Comment:
shouldn't this an error?  
I think user should be alerted (with error) that the chainId requested is not staked by that provider.